test/e2e/framework/consts.go

```diff
@@ -31,7 +31,7 @@ const (
 var (
 	// SvcExportConditionCmpOptions configures comparison behaviors foo service export conditions.
 	SvcExportConditionCmpOptions = []cmp.Option{
-		cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime", "ObservedGeneration", "Message"),
+		cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime", "Message"),
```

Comment:
why remove this? I thought the whole PR is about setting it

Reply:
It's used when we compare the serviceExport condition.
Before we ignored the ObservedGeneration.